n3fit/src/n3fit/model_trainer.py

```diff
@@ -523,12 +522,43 @@ def _generate_observables(
         self._reset_observables()
         log.info("Generating layers")
 
+        # We need to transpose Experimental data, stacking over replicas
+        experiment_data = {
+            "trmask": [],
+            "expdata": [],
+            "expdata_vl": [],
+            "invcovmat": [],
+            "invcovmat_vl": [],
+        }
+
+        # Loop over datasets
+        for i in range(len(self.exp_info[0])):
+            # Loop over data fields
+            for key in experiment_data.keys():
+                replica_data = []
+                # Loop over replicas
+                for replica in self.exp_info:
+                    if key in ["expdata", "expdata_vl"]:
+                        replica_data.append(replica[i][key].flatten())
+                    else:
+                        replica_data.append(replica[i][key])
+                # Stack
+                experiment_data[key].append(np.stack(replica_data))
+
         # Now we need to loop over all dictionaries (First exp_info, then pos_info and integ_info)
-        for exp_dict in self.exp_info:
+        for i in range(len(self.exp_info[0])):
```

Comment:
```suggestion
        for i, exp_dict in enumerate(self.exp_info[0]):
```

Since you are using explicitly `self.exp_info[0][i]` below.

Reply:
Fixed in [2bbf864](https://github.com/NNPDF/nnpdf/pull/1788/commits/2bbf86457057227ecba60e6d8da9473f2f1b3279)